lib/utils.js

```diff
@@ -1,53 +1,48 @@
-var Promise = window.Promise || require('../vendor/native-promise-only');
-
-function isObject(obj) {
-  return (obj !== null && typeof obj === 'object');
+export function isObject (obj) {
+  return (obj !== null && typeof obj === 'object')
 }
 
-function decode(s) {
-  return decodeURIComponent((s || '').replace( /\+/g, " " ));
+function decode (s) {
+  return decodeURIComponent((s || '').replace(/\+/g, ' '))
 }
 
-function queryParameters(queryString) {
-  queryString = queryString || '';
+export function queryParameters (queryString) {
+  queryString = queryString || ''
 
-  var result = {},
-      keyValuePairs,
-      keyAndValue,
-      key,
-      value;
+  const result = {}
+  let keyValuePairs, keyAndValue, key, value
 
   if (queryString.search(/\?|#/) === 0) {
-    queryString = queryString.slice(1);
+    queryString = queryString.slice(1)
   }
 
-  if (queryString.length === 0) { return result; }
+  if (queryString.length === 0) { return result }
 
-  keyValuePairs = queryString.split('&');
+  keyValuePairs = queryString.split('&')
 
-  for (var i = 0; i < keyValuePairs.length; i++) {
-    keyAndValue = keyValuePairs[i].split('=');
-    key   = decode(keyAndValue[0]);
-    value = decode(keyAndValue[1]) || '';
-    result[key] = value;
+  for (let i = 0; i < keyValuePairs.length; i++) {
```

Comment:
linter will not allow you to do this `for (let i = 0; i < keyValuePairs.length; i++) {`.

It will complain that the variable is never reassigned, please use `const`.

Reply:
`i++` ?

Reply:
Ops. I think Mehul and I got confused. We can't use const when iterating with this old school for loop

HOWEVER, the other for loops. I believe we were confused for in and for of.